custom_components/ramses_cc/__init__.py

```diff
@@ -207,38 +206,16 @@ def extra_state_attributes(self) -> dict[str, Any]:
     async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
         self._broker._entities[self.unique_id] = self
-        async_dispatcher_connect(self.hass, DOMAIN, self.async_handle_dispatch)
-
-    @callback  # TODO: WIP
-    def _call_client_api(self, func, *args, **kwargs) -> None:
-        """Wrap client APIs to make them threadsafe."""
-        # self.hass.loop.call_soon_threadsafe(
-        #     func(*args, **kwargs)
-        # )  # HACK: call_soon_threadsafe should not be needed
-
-        func(*args, **kwargs)
-        self.update_ha_state()
-
-    @callback
-    def async_handle_dispatch(self, *args) -> None:  # TODO: remove as unneeded?
-        """Process a dispatched message.
-
-        Data validation is not required, it will have been done upstream.
-        This routine is threadsafe.
-        """
-        if not args:
-            self.update_ha_state()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIGNAL_UPDATE, self.async_write_ha_state
+            )
+        )
 
     @callback
-    def update_ha_state(self, delay=3) -> None:
-        """Update HA state after a short delay to allow system to quiesce.
-
-        This routine is threadsafe.
-        """
-        args = (delay, self.async_schedule_update_ha_state)
-        self.hass.loop.call_soon_threadsafe(
-            self.hass.helpers.event.async_call_later, *args
-        )  # HACK: call_soon_threadsafe should not be needed
```

Comment:
So, IIRC, _should_ not be needed, but _is_ (well, was)... Have you tested this?

We have no test suite. :(